src/Featurizers/Components/RollingWindowTransformerBase.h

```diff
@@ -0,0 +1,163 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "Calculators/Calculators.h"
+#include "Components/InferenceOnlyFeaturizerImpl.h"
+#include "Components/WindowFeaturizerBase.h"
+#include "../Featurizer.h"
+#include "../Traits.h"
+
+#include <vector>
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         RollingWindowTransformerBase
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///                 This class does not have an archive constructor or a save method as those need
+///                 to be implemented by the derived classes.
+///
+template <
+    typename InputT,
+    typename OutputT,
+    size_t MaxNumTrainingItemsV
+>
+class RollingWindowTransformerBase:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>>;
+    using CalculatorFunctionInputType = typename Components::CircularBuffer<InputT>::iterator;
+    using CalculatorFunction = std::function<typename OutputT(typename CalculatorFunctionInputType, typename CalculatorFunctionInputType)>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    RollingWindowTransformerBase(std::uint32_t windowSize, CalculatorFunction calculator, std::uint32_t horizon, std::uint32_t minWindowSize) ;
+    ~RollingWindowTransformerBase(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(RollingWindowTransformerBase);
+
+    bool operator==(RollingWindowTransformerBase const &other) const;
+    bool operator!=(RollingWindowTransformerBase const &other) const;
+
+protected:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Protected Members
+    // |
+    // ----------------------------------------------------------------------
+    // These are protected so that the derived classes can use them for saving of state.
+    const std::uint32_t                             _windowSize;

```

Comment:
>_windowSize [](start = 52, length = 11)

I'd suggest renaming this to `_maxWindowSize`

Reply:
Renamed.